framework_tvb/tvb/interfaces/web/controllers/simulator/monitors_wizard_handler.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+.. moduleauthor:: Robert Vincze <robert.vincze@codemart.ro>
+"""
+
+from tvb.adapters.simulator.equation_forms import get_form_for_equation
+from tvb.adapters.simulator.monitor_forms import AdditiveNoiseViewModel, get_ui_name_to_monitor_dict, \
+    get_form_for_monitor, prepare_monitor_legend, BoldViewModel
+from tvb.adapters.simulator.simulator_fragments import SimulatorMonitorFragment, SimulatorFinalFragment
+from tvb.core.services.algorithm_service import AlgorithmService
+from tvb.interfaces.web.controllers.simulator.simulator_wizzard_urls import SimulatorWizzardURLs
+
+
+class MonitorsWizardHandler:
+    def __init__(self):
+        self.next_monitors_dict = None
+
+    def set_monitors_list_on_simulator(self, session_stored_simulator, monitor_names):
+        self.build_list_of_monitors_from_names(monitor_names, session_stored_simulator.is_surface_simulation)
+        session_stored_simulator.monitors = list(monitor for monitor, _ in self.next_monitors_dict.values())
+
+    def clear_next_monitors_dict(self):
+        if self.next_monitors_dict:
+            self.next_monitors_dict.clear()
+
+    def build_list_of_monitors_from_view_models(self, monitor_view_models):
+        self.next_monitors_dict = dict()
+        count = 0
+        for monitor_model in monitor_view_models:
+            self.next_monitors_dict[monitor_model.__class__.__name__] = (monitor_model, count + 1)
+            count = count + 1
```

Comment:
Looking at the usages of the `next_monitors_dict` it seems we could simplify it to hold only classname -> index pairs. What do you think?